cpp/src/arrow/util/bitmap_ops.cc

```diff
@@ -215,17 +215,19 @@ void ReverseBlockOffsets(const uint8_t* data, int64_t offset, int64_t length,
 
 template <TransferMode mode>
 Result<std::shared_ptr<Buffer>> TransferBitmap(MemoryPool* pool, const uint8_t* data,
-                                               int64_t offset, int64_t length) {
-  ARROW_ASSIGN_OR_RAISE(auto buffer, AllocateEmptyBitmap(length, pool));
+                                               int64_t offset, int64_t length,
+                                               int64_t out_offset) {
+  const int64_t phys_bits = length + out_offset;
+  ARROW_ASSIGN_OR_RAISE(auto buffer, AllocateEmptyBitmap(phys_bits, pool));
   uint8_t* dest = buffer->mutable_data();
 
-  TransferBitmap<mode>(data, offset, length, 0, dest);
+  TransferBitmap<mode>(data, offset, length, out_offset, dest);
 
   // As we have freshly allocated this bitmap, we should take care of zeroing the
   // remaining bits.
```

Comment:
It is actually dubious that this code is necessary, since `AllocateEmptyBitmap` has already zero-initialized the entire buffer. Perhaps we can try remove it?

Reply:
I've tried removing it and all tests are successful